Contact/Validator.cs

```diff
@@ -24,7 +24,7 @@ public static bool hasValidEmail( this Contact contact )
         {

```

Comment:
this approach is ok but you might do some more research on model validation in asp.net core as there are other techniques.

Reply:
Okay, this will go on my to-do list.

Reply:
I'm starting to put this together but may take another day on this.